spec/rspec/support/caller_filter_spec.rb

```diff
@@ -43,6 +43,14 @@ def unmatched_from(files)
         expect(unmatched_from files).to eq(files)
       end
 
+      it "matches internal methods" do
+        files = [
+          "<internal:kernel>:90:in `tap'",
+          "<internal:core> core/kernel.rb:534:in `tap'"
```

Comment:
This is not a real world integration example, I want to see a scenario that generates these lines on C ruby.

Reply:
Would something like
```ruby
expect(tap { RSpec::CallerFilter.first_non_rspec_line }).to include("#{__FILE__}:#{__LINE__}")
```
be OK?

Reply:
```ruby
    # Needs CRuby 3.0 or TruffleRuby to have <internal: entries
    it 'returns the caller ignoring <internal: entries from the core library in the backtrace' do
      # Skip locations inside the block of #tap as we want to test
      # as if the beginning of the backtrace is in Kernel#tap
      location = nil
      skip = 0
      line = __LINE__ + 2
      begin
        tap { location = RSpec::CallerFilter.first_non_rspec_line(skip) }
        skip += 1
      end while location.include?('block (3 levels)')
      expect(location).to include("#{__FILE__}:#{line}:in `block (2 levels)")
      expect(location).not_to include('<internal:')
    end
```
This fails on CRuby master without this PR's fix.
@bjfish Can you add it? (around `spec/rspec/support/caller_filter_spec.rb:15`)

Reply:
@JonRowe I've added the test provided by @eregon above.  The tests mentioned here: https://github.com/rspec/rspec-support/pull/426#issuecomment-693247485 also cover this.

I limited the scope of implementation/version based on the results I saw while running the CI: https://travis-ci.org/github/bjfish/rspec-support/builds/727733143